File: scripts/algorithms/E/Element Appearing More Than 25% In Sorted Array/Element Appearing More Than 25% In Sorted Array.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        l=len(arr)
        c=(l//4)+1
        d={}
        for i in arr:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
            if d[i]>=c:
                return i

File: scripts/algorithms/E/Element Appearing More Than 25 In Sorted Array/Element Appearing More Than 25% In Sorted Array.cpp
// Runtime: 45 ms (Top 0%) | Memory: 14.4 MB (Top 0%)
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {

        int freq = 0.25 * arr.size();
        map<int,int>m;
        for(int i: arr)
            m[i]++;

        int k;
        for(auto i: m)
        {
            if(i.second > freq)
            {
              k = i.first;
                break;
            }
        }
        return k;
    }
};